src/types.ts

```diff
@@ -68,6 +68,15 @@ export interface ActionPluginPayload {
   required_parameters: { [key: string]: any }
 }
 
+export interface User {
+  first_name?: string
+  last_name?: string
+  username: string
+  date_of_birth?: Date
+  password_hash: string
+  email: string
+}
+
```

Comment:
I think it might be better to use the types from the generated `schema.graphql` that was included in the previous version of this postgraphile server. I think using this would be better than redefining the types here.

Reply:
Yeah, good point -- have you imported any of the `schema.graphql` types elsewhere, just so I can see how it's done?

Just one problem though maybe -- will the generated types file know which fields should be optional? (e.g. `date_of_birth?: Date`)

Reply:
I haven't used here in the back-end, just front-end. Which is done via Graphql `code-gen` package. But it should be similar here. When the postgraphile runs it creates this file and it contains all these types in there.
Ah, I don't know how to do this optional with the existing types... There is probably a way to define a type a make one or more fields optional.